cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -312,11 +324,33 @@ func (scaleSet *ScaleSet) setScaleSetSize(size int64, delta int) error {
 	if requiredInstances > 0 {
 		klog.V(3).Infof("Remaining unsatisfied count is %d. Attempting to increase scale set %q "+
 			"capacity", requiredInstances, scaleSet.Name)
-		err := scaleSet.createOrUpdateInstances(&vmssInfo, size)
+		future, err := scaleSet.createOrUpdateInstances(&vmssInfo, size)
 		if err != nil {
+			klog.Errorf("Failed to request capacity for scale set %q to %d: %v", scaleSet.Name, requiredInstances, err)
+			return err
+		}
+
+		ctx, cancel := getContextWithTimeout(syncContextTimeout)
+		defer cancel()
+		klog.V(3).Infof("Waiting %s for scaleSet.waitForCreateOrUpdateInstances(%s)", syncContextTimeout, scaleSet.Name)
+		err = scaleSet.waitForCreateOrUpdateInstances(ctx, false, future)
+		if err == nil {
+			klog.V(3).Infof("Increased capacity for scale set %q to %d", scaleSet.Name, requiredInstances)
+			return nil
+		}
+
+		if !errors.Is(err, context.DeadlineExceeded) {
 			klog.Errorf("Failed to increase capacity for scale set %q to %d: %v", scaleSet.Name, requiredInstances, err)
 			return err
 		}
+
+		// Switch to async wait for the outcome of VMSS capacity update.
+		klog.V(3).Infof("Waiting %s for scaleSet.waitForCreateOrUpdateInstances(%s) in async mode", asyncContextTimeout, scaleSet.Name)
```

Comment:
preference for not logging this, or changing the language (we aren't actually waiting in this flow)

Reply:
Actually removed both comments.